Fix an HDF5 test

We have a test that uses an HDF5 function that had previously been removed:
http://cdash.kyomu.43-1.org/testDetails.php?test=6728357&build=4094

The error message goes like this:
```
/mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc: In instantiation of ‘void test() [with int dim = 2]’:
/mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:77:15:   required from here
/mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:52:3: error: no matching function for call to ‘dealii::DataOut<2, dealii::DoFHandler<2> >::write_hdf5_parallel(const char [7], MPI_Comm)’
   data_out.write_hdf5_parallel ("out.h5", MPI_COMM_WORLD);
   ^
/mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:52:3: note: candidates are:
In file included from /mnt/data/testsuite/dealii/include/deal.II/numerics/data_out_dof_data.h:23:0,
                 from /mnt/data/testsuite/dealii/include/deal.II/numerics/data_out.h:22,
                 from /mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:24:
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2379:8: note: void dealii::DataOutInterface<dim, spacedim>::write_hdf5_parallel(const dealii::DataOutBase::DataOutFilter&, const string&, MPI_Comm) const [with int dim = 2; int spacedim = 2; std::string = std::basic_string<char>; MPI_Comm = ompi_communicator_t*]
   void write_hdf5_parallel (const DataOutBase::DataOutFilter &data_filter,
        ^
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2379:8: note:   candidate expects 3 arguments, 2 provided
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2390:8: note: void dealii::DataOutInterface<dim, spacedim>::write_hdf5_parallel(const dealii::DataOutBase::DataOutFilter&, bool, const string&, const string&, MPI_Comm) const [with int dim = 2; int spacedim = 2; std::string = std::basic_string<char>; MPI_Comm = ompi_communicator_t*]
   void write_hdf5_parallel (const DataOutBase::DataOutFilter &data_filter,
        ^
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2390:8: note:   candidate expects 5 arguments, 2 provided
/mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:54:3: error: no matching function for call to ‘dealii::DataOut<2, dealii::DoFHandler<2> >::create_xdmf_entry(const char [7], int, MPI_Comm)’
   xdmf_entries.push_back(
   ^
/mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:54:3: note: candidates are:
In file included from /mnt/data/testsuite/dealii/include/deal.II/numerics/data_out_dof_data.h:23:0,
                 from /mnt/data/testsuite/dealii/include/deal.II/numerics/data_out.h:22,
                 from /mnt/data/testsuite/dealii/tests/mpi/data_out_hdf5_01.cc:24:
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2326:13: note: dealii::XDMFEntry dealii::DataOutInterface<dim, spacedim>::create_xdmf_entry(const dealii::DataOutBase::DataOutFilter&, const string&, double, MPI_Comm) const [with int dim = 2; int spacedim = 2; std::string = std::basic_string<char>; MPI_Comm = ompi_communicator_t*]
   XDMFEntry create_xdmf_entry (const DataOutBase::DataOutFilter &data_filter,
             ^
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2326:13: note:   candidate expects 4 arguments, 3 provided
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2336:13: note: dealii::XDMFEntry dealii::DataOutInterface<dim, spacedim>::create_xdmf_entry(const dealii::DataOutBase::DataOutFilter&, const string&, const string&, double, MPI_Comm) const [with int dim = 2; int spacedim = 2; std::string = std::basic_string<char>; MPI_Comm = ompi_communicator_t*]
   XDMFEntry create_xdmf_entry (const DataOutBase::DataOutFilter &data_filter,
             ^
/mnt/data/testsuite/dealii/include/deal.II/base/data_out_base.h:2336:13: note:   candidate expects 5 arguments, 3 provided
```
This needs to be adjusted.

--- a/tests/mpi/data_out_hdf5_01.cc
+++ b/tests/mpi/data_out_hdf5_01.cc
@@ -49,10 +49,13 @@ test ()
   data_out.add_data_vector (dof1, v1, "linear");
   data_out.build_patches (2);
 
-  data_out.write_hdf5_parallel ("out.h5", MPI_COMM_WORLD);
+  DataOutBase::DataOutFilter data_filter
+  (DataOutBase::DataOutFilterFlags (false, false));
+  data_out.write_filtered_data (data_filter);
+  data_out.write_hdf5_parallel (data_filter, "out.h5", MPI_COMM_WORLD);
   std::vector<XDMFEntry> xdmf_entries;
-  xdmf_entries.push_back(
-    data_out.create_xdmf_entry("out.h5", 0, MPI_COMM_WORLD));
+  xdmf_entries.push_back
+  (data_out.create_xdmf_entry (data_filter, "out.h5", 0, MPI_COMM_WORLD));
 
   data_out.write_xdmf_file(xdmf_entries, "out.xdmf",
 			   MPI_COMM_WORLD);

--- a/tests/mpi/data_out_hdf5_01.cc
+++ b/tests/mpi/data_out_hdf5_01.cc
@@ -34,8 +34,6 @@ test ()
   Triangulation<dim> tria;
   GridGenerator::hyper_cube(tria, 0., 1.);
   tria.refine_global (1);
-  //  tria.begin_active()->set_refine_flag();
-  //tria.execute_coarsening_and_refinement ();
 
   FE_Q<dim> fe1(1);
 
@@ -57,16 +55,15 @@ test ()
   xdmf_entries.push_back
   (data_out.create_xdmf_entry (data_filter, "out.h5", 0, MPI_COMM_WORLD));
 
-  data_out.write_xdmf_file(xdmf_entries, "out.xdmf",
-			   MPI_COMM_WORLD);
+  data_out.write_xdmf_file (xdmf_entries, "out.xdmf", MPI_COMM_WORLD);
 
-  deallog <<"ok" << std::endl;
-  
-  
+  deallog << "ok" << std::endl;
+
+
   // Sadly hdf5 is binary and we can not use hd5dump because it might
   // not be in the path. At least we can look at the xdmf:
   if (0==Utilities::MPI::this_mpi_process (MPI_COMM_WORLD))
-    cat_file("out.xdmf");  
+    cat_file("out.xdmf");
 }
 
 
@@ -78,7 +75,6 @@ int main(int argc, char *argv[])
   try
     {
       test<2>();
-//      test<3>();
 
       return 0;
     }
@@ -106,4 +102,3 @@ int main(int argc, char *argv[])
       return 1;
     };
 }
-

